examples/components/CCFRAUD_VERTICAL/traininsilo/contributor.py

```diff
@@ -0,0 +1,404 @@
+        test_data_dir="./",
+        model_path=None,
+        lr=0.01,
+        epochs=1,
+        batch_size=10,
+        experiment_name="default-experiment",
+        iteration_name="default-iteration",
+    ):
+        """Credit Card Fraud Trainer trains simple model on the Fraud dataset.
+
+        Args:
+            model_name(str): Name of the model to use for training, options: SimpleLinear, SimpleLSTM, SimpleVAE
+            train_data_dir(str, optional): Training data directory path
+            test_data_dir(str, optional): Testing data directory path
+            lr (float, optional): Learning rate. Defaults to 0.01
+            epochs (int, optional): Epochs. Defaults to 1
+            batch_size (int, optional): DataLoader batch size. Defaults to 64.
+
+        Attributes:
+            model_: Model
+            criterion_: BCELoss loss
+            optimizer_: Stochastic gradient descent
+            train_dataset_: Training Dataset obj
+            train_loader_: Training DataLoader
+            test_dataset_: Testing Dataset obj
+            test_loader_: Testing DataLoader
+        """
+
+        # Training setup
+        self._lr = lr
+        self._epochs = epochs
+        self._batch_size = batch_size
+        self._experiment_name = experiment_name
+        self._iteration_name = iteration_name
+        self._global_rank = global_rank
+        self._global_size = global_size
+        self._global_comm = global_comm
+
+        self.device_ = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        logger.info(f"Using device: {self.device_}")
+
+        self.train_dataset_, self.test_dataset_, self._input_dim = self.load_dataset(
+            train_data_dir, test_data_dir, model_name
+        )
+        self.train_loader_ = DataLoader(
+            self.train_dataset_, batch_size=batch_size, shuffle=False
+        )
+        self.test_loader_ = DataLoader(
+            self.test_dataset_, batch_size=batch_size, shuffle=False
+        )
+
+        # Build model
+        self._model_name = model_name
+        if self._global_rank == 0:
+            self.model_ = getattr(models, model_name + "Top")().to(self.device_)
+        else:
+            self.model_ = getattr(models, model_name + "Bottom")(self._input_dim).to(
+                self.device_
+            )
+        self._model_path = model_path
+
+        self.criterion_ = nn.BCELoss()
+        self.optimizer_ = SGD(self.model_.parameters(), lr=self._lr, weight_decay=1e-5)
+
+    def load_dataset(self, train_data_dir, test_data_dir, model_name):
+        """Load dataset from {train_data_dir} and {test_data_dir}
+
+        Args:
+            train_data_dir(str): Training data directory path
+            test_data_dir(str): Testing data directory path
+            model_name(str): Name of the model to use
+        """
+        logger.info(f"Train data dir: {train_data_dir}, Test data dir: {test_data_dir}")
+        if self._global_rank == 0:
```

Comment:
same here, looks like this test on global_rank is a remaining code from before the host and contributor scripts were separated?